src/components/checkbox/checkbox.js

```diff
@@ -27,6 +27,16 @@ angular
  * @param {string=} ng-change Angular expression to be executed when input changes due to user interaction with the input element.
  * @param {boolean=} md-no-ink Use of attribute indicates use of ripple ink effects
  * @param {string=} aria-label Adds label to checkbox for accessibility.
+ * @param {expression=} indeterminate This determines when the checkbox
+ * should be rendered as 'indeterminate'.
+ * If a truthy expression or no value is passed in the checkbox renders in the indeterminate state.
+ * If falsy expression is passed in it just looks like a normal unchecked checkbox.
+ * To use the other indeterminate-* attributes this attribute must be present.
+ * @param {expression=} indeterminate-checked-when This determines when the
+ * indeterminate checkbox should appear checked. The 'indeterminate' state is mutually
+ * exclusive with the 'checked' and 'unchecked' states.
+ * @param {expression=} indeterminate-click This gets triggered when the
```

Comment:
I do not understand the feature here. At first glance, this seems to interfere with ng-click. Or is this an expression that should be invoked with the indeterminant state changes?

e.g.  `md-change-indeterminant`

What is the use case here ?